packages/amplify-category-api/src/provider-utils/awscloudformation/utils/rds-secrets/ssmClient.ts

```diff
@@ -0,0 +1,115 @@
+import { $TSAny, $TSContext } from 'amplify-cli-core';
+import aws from 'aws-sdk';
+
+/**
+ *  SSM client provider for AWS SDK calls
+ */
+export class SSMClient {
+  private static instance: SSMClient;
+
+  static getInstance = async (context: $TSContext): Promise<SSMClient> => {
+    if (!SSMClient?.instance) {
+      SSMClient.instance = new SSMClient(await getSSMClient(context));
```

Comment:
Does the region need to be yanked from the client context, or is it set globally somewhere in the CLI?

Reply:
The region is available from the context and is appropriately resolved by the provider plugin before it creates the client.